modin/backends/pandas/query_compiler.py

```diff
@@ -2620,17 +2657,76 @@ def compute_groupby(df):
+                            for o in by
+                            if isinstance(o, str)
+                        )
+
+                        if not has_categorical_by:
+                            # Deleting index levels that already exists in the result frame
+                            duplicated_cols = [
+                                i
+                                for i, name in enumerate(result.index.names)
+                                if name in result.columns
+                                or (
+                                    isinstance(name, str)
+                                    and name.startswith("__reduced__")
+                                )
+                            ]
+                            if len(duplicated_cols) == result.index.nlevels:
+                                drop = False
+                            else:
+                                result.index = result.index.droplevel(duplicated_cols)
+                        else:
+                            # Deleting columns that already exists in the result index levels
+                            duplicated_cols = [
+                                name
+                                for name in result.index.names
+                                if name in result.columns
+                            ]
+                            result.drop(columns=duplicated_cols, inplace=True)
+                    result.reset_index(drop=not drop, inplace=True)
+
+                new_index_names = [
+                    None
+                    if isinstance(name, str) and name.startswith("__reduced__")
+                    else name
+                    for name in result.index.names
+                ]
+
+                cols_to_drop = (
+                    result.columns[result.columns.str.match(r"__reduced__.*")]
+                    if hasattr(result.columns, "str")
+                    else []
+                )
+
+                result.index.names = new_index_names
+
+                # Not dropping columns if result is Series
+                if len(result.columns) > 1:
+                    result.drop(columns=cols_to_drop, inplace=True)
+
                 return result
 
             try:
-                return compute_groupby(df)
+                return compute_groupby(df, drop)
             # This will happen with Arrow buffer read-only errors. We don't want to copy
             # all the time, so this will try to fast-path the code first.
             except (ValueError, KeyError):
-                return compute_groupby(df.copy())
+                return compute_groupby(df.copy(), drop)
 
-        new_modin_frame = self._modin_frame._apply_full_axis(
-            axis, lambda df: groupby_agg_builder(df)
+        new_modin_frame = self._modin_frame.broadcast_apply_full_axis(
+            axis=axis,
+            func=lambda df, by=None: groupby_agg_builder(df, by, drop),
```

Comment:
Is `by` always passed to `groupby_agg_builder` as `None`?

Reply:
No, depends on is `len(broadcastable_by) == 0` or not, `func` will obtain either `None` or broadcasted frame